File: Front/ProEventos-App/src/app/models/Palestrante.ts
import {UserUpdate} from './Identity/UserUpdate';
import { Evento } from './Evento';
import { RedeSocial } from './RedeSocial';

export interface Palestrante {
  id: number;
  miniCurriculo: string;
  user: UserUpdate;
  redesSociais: RedeSocial[];
  eventos: Evento[];
}

File: Back/ProEventos/ProEventos.Domain/Models/Palestrante.cs
using System.Collections;
using System.Collections.Generic;
using ProEventos.Domain.Identity;

namespace ProEventos.Domain.Models
{
    public class Palestrante
    {
        public int Id { get; set; }        

        public string MiniCurriculo { get; set; }

        public int UserId { get; set; }

        public User User { get; set; }

        public IEnumerable<RedeSocial> RedesSociais { get; set; }

        public IEnumerable<Evento> Eventos { get; set; }
    }
}